codersdk/provisionerdaemons.go

```diff
@@ -198,6 +200,7 @@ func (c *Client) ServeProvisionerDaemon(ctx context.Context, req ServeProvisione
 	}
 	query := serverURL.Query()
 	query.Add("id", req.ID.String())
+	query.Add("name", req.Name)
```

Comment:
curious: is it required to pass ID and name now?

Reply:
I think I'm going to end up ignoring the ID parameter and just upserting based on name in a follow-up PR.